diff --git a/dist/index.cjs b/dist/index.cjs
index 0f7becca99bd84dfd6bbc0544d5d9e03f36220c4..e9ce4a49bb48972163737f084684dbf03906c3fd 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -5023,13 +5023,13 @@ function deserializeWithBigInt(serialized) {
     return v;
   });
 }
-var storage = persistenceConfig.storage === "localStorage" ? window.localStorage : window.sessionStorage;
-var persister = (0, import_query_async_storage_persister.createAsyncStoragePersister)({
+var storage = typeof window === "undefined" ? void 0 : persistenceConfig.storage === "localStorage" ? window.localStorage : window.sessionStorage;
+var persister = storage ? (0, import_query_async_storage_persister.createAsyncStoragePersister)({
   storage,
   key: persistenceConfig.key ?? "cofhe:react-query",
   serialize: serializeWithBigInt,
   deserialize: deserializeWithBigInt
-});
+}) : void 0;
 
 // src/providers/internalQueryHooks.ts
 function applyPersistedDefaults(options) {
diff --git a/dist/index.js b/dist/index.js
index f6967766b628e2140e0cf57feab8f861fa7a1d89..e632c558fc7ce816ad371ba1f61312ea60013a16 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -4964,13 +4964,13 @@ function deserializeWithBigInt(serialized) {
     return v;
   });
 }
-var storage = persistenceConfig.storage === "localStorage" ? window.localStorage : window.sessionStorage;
-var persister = createAsyncStoragePersister({
+var storage = typeof window === "undefined" ? void 0 : persistenceConfig.storage === "localStorage" ? window.localStorage : window.sessionStorage;
+var persister = storage ? createAsyncStoragePersister({
   storage,
   key: persistenceConfig.key ?? "cofhe:react-query",
   serialize: serializeWithBigInt,
   deserialize: deserializeWithBigInt
-});
+}) : void 0;
 
 // src/providers/internalQueryHooks.ts
 function applyPersistedDefaults(options) {
